src/graphs.jl

```diff
@@ -306,6 +307,11 @@ end
     if get(plotattributes, :linewidth, 1) > 0
         # generate a list of colors, one per segment
         segment_colors = get(plotattributes, :linecolor, nothing)
+        edge_label_array = Vector{Tuple}()
+        if typeof(edgelabel) <: AbstractArray
+            edgelabel = vec(edgelabel)
+            (edge_label_array_size = round(Int, sqrt(length(edgelabel))))
```

Comment:
Why is this in brackets? And what's the thinking exactly?

Reply:
The brackets are a mistake from some sort of copy/paste job. This piece of code makes it so that the user can pass in a matrix and the code will still work. Now that I think of it, I think that we should go the other way round. If the user passes in a matrix, then we don't do any conversion and if they pass in a vector, then we `reshape` into a matrix. That way I wouldn't have to do any of the confusing `LinearIndices` business that I go through on lines 389 and 393.